megamek/src/megamek/utilities/NameChangesValidator.java

```diff
@@ -62,33 +65,57 @@ public static void main(String... args) {
         validator.testRightSide();
     }
 
-    private void testEqualSides() {
-        // Find equal left and right sides
-        logger.info("Looking for equal left and right sides...");
-        try (FileInputStream fis = new FileInputStream(lookupNames);
-                InputStreamReader isr = new InputStreamReader(fis, StandardCharsets.UTF_8);
-                BufferedReader br = new BufferedReader(isr)) {
-            String line;
-            while (null != (line = br.readLine())) {
+    private List<String> loadFile(File fileName) {
+        String message = String.format("Collecting lines from file %s", fileName);
+        logger.info(message);
+
+        List<String> lines = new ArrayList<>();
+
+        try {
+            FileInputStream fis = new FileInputStream(fileName);
+            InputStreamReader isr = new InputStreamReader(fis, StandardCharsets.UTF_8);
+            BufferedReader br = new BufferedReader(isr);
+
+            String line = "";
+
+            while ((line = br.readLine()) != null) {
                 if (line.startsWith("#")) {
                     continue;
                 }
-                int index = line.indexOf('|');
-                if (index > 0) {
-                    String lookupName = line.substring(0, index);
-                    String entryName = line.substring(index + 1);
-                    if (lookupName.equals(entryName)) {
-                        String message = String.format("Equal lookup name and cache entry in line: %s", line);
-                        logger.info(message);
-                        errors++;
-                    }
-                }
+
+                lines.add(line);
             }
-        } catch (Exception ex) {
-            logger.error(ex, STRING_EXCEPTION);
+
+            br.close();
```

Comment:
Do we need to close the reader in the catches too? As (unless I'm thinking about things wrong) wouldn't we get a memory leak if the streamed resources were left dangling?

Reply:
Utility exists if there is an exception so it'll close automatically.